spec.js

```diff
@@ -182,29 +176,29 @@ Spec.prototype.compileEnum = function compileEnum(def) {
     var spec = new EnumSpec();
     spec.compile(def, self);
     self.claim(spec.name, def.id);
-    self.types[spec.name] = spec;
+    self.typeSpecs[spec.name] = spec;
 };
 
 Spec.prototype.link = function link() {
     var self = this;
     var index;
 
-    var typeNames = Object.keys(self.types);
+    var typeNames = Object.keys(self.typeSpecs);
     for (index = 0; index < typeNames.length; index++) {
-        var type = self.types[typeNames[index]];
-        type.link(self);
+        var type = self.typeSpecs[typeNames[index]];
+        self[type.name] = type.link(self).surface;
     }
 
-    var serviceNames = Object.keys(self.services);
+    var serviceNames = Object.keys(self.serviceSpecs);
     for (index = 0; index < serviceNames.length; index++) {
-        var service = self.services[serviceNames[index]];
-        service.link(self);
+        var service = self.serviceSpecs[serviceNames[index]];
+        self[service.name] = service.link(self).surface;
     }
 
     var constNames = Object.keys(self.constSpecs);
     for (index = 0; index < constNames.length; index++) {
         var constSpec = self.constSpecs[constNames[index]];
-        self.consts[constNames[index]] = constSpec.link(self);
+        self[constSpec.name] = constSpec.link(self).surface;
```

Comment:
This function is the meat of this change. All properties of the thrift spec get assigned here.